sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -0,0 +1,455 @@
+
+                score_breakdown = self._get_score_breakdown(
+                    metric,
+                    column_name_1,
+                    column_name_2,
+                    processed_real_data,
+                    discrete_real,
+                    processed_synthetic_data,
+                    discrete_synthetic,
+                    metadata,
+                )
+                pair_score = score_breakdown['score']
+                if metric.__name__ == 'CorrelationSimilarity':
+                    real_correlation = score_breakdown['real']
+                    synthetic_correlation = score_breakdown['synthetic']
+                else:
+                    real_correlation = np.nan
+                    synthetic_correlation = np.nan
+
+            except Exception as e:
+                pair_score = np.nan
+                real_correlation = np.nan
+                synthetic_correlation = np.nan
+                if not str(e) == 'Preprocessing failed':
+                    error = f'Error: {type(e).__name__} {e}'
+
+            column_names_1.append(column_name_1)
+            column_names_2.append(column_name_2)
+            metric_names.append(metric.__name__)
+            scores.append(pair_score)
+            real_correlations.append(real_correlation)
+            synthetic_correlations.append(synthetic_correlation)
+            error_messages.append(error)
+
+            if progress_bar:
+                progress_bar.update()
+
+        result = pd.DataFrame({
+            'Column 1': column_names_1,
+            'Column 2': column_names_2,
+            'Metric': metric_names,
+            'Score': scores,
+            'Real Correlation': real_correlations,
+            'Synthetic Correlation': synthetic_correlations,
+            'Error': error_messages
+        })
+
+        if result['Error'].isna().all():
+            result = result.drop('Error', axis=1)
+
+        return result
+
+    def _get_correlation_matrix(self, column_name):
+        """Get the correlation matrix for the given column name."""
+        if column_name not in ['Score', 'Real Correlation', 'Synthetic Correlation']:
+            raise ValueError(f"Invalid column name for _get_correlation_matrix : '{column_name}'")
+
+        table = self._details.dropna(subset=[column_name])
+        names = list(pd.concat([table['Column 1'], table['Column 2']]).unique())
+        heatmap_df = pd.DataFrame(index=names, columns=names)
+
+        for idx_1, column_name_1 in enumerate(names):
+            for column_name_2 in names[idx_1:]:
+                if column_name_1 == column_name_2:
+                    heatmap_df.loc[column_name_1, column_name_2] = 1
+                    continue
+
+                col_1_loc = (table['Column 1'] == column_name_1)
+                col_2_loc = (table['Column 2'] == column_name_2)
+                if table.loc[col_1_loc & col_2_loc].empty:
+                    col_1_loc = (table['Column 1'] == column_name_2)
```

Comment:
Can you leave a small comment here about why we check both combinations?

Reply:
Yes I let a comment lint 324-325